### ISSUE TITLE
E0 will not drive extruder stepper 

### ISSUE THREAD

**Issue by SpectrumTech** [bigtreetech/BIGTREETECH-SKR-E3-Turbo#27]

**Comment by SpectrumTech** [comment]
BIGTREE_TECH is not being helpful at all on this. They keep wanting me to troubleshoot the problem myself. Now they want me to send them a video of the problem - which is exactly as I describe. So I really don't get the point. With the size of this printer and the way the enclosure for the electronics is made it is hard to shoot a video while handling a camera on top of changing settings, not to mention dangerous because of all the mains power exposed in the box. so I am just going to return the board while the return window is open and be done with BIGTREE_TECH.

**Comment by stsweeney41** [comment]
I agree, there is something wrong between this board and the marlin software. I am going to try  some other firmware as a test and see if that solves the problem.

**Comment by sgsrtcameron** [comment]
I am having the same issue right now, did you ever find a solution?

**Comment by stsweeney41** [comment]
Unfortunately no... may have to go with a board that has Plug in drivers so
that it is just standalone.

**Comment by sgsrtcameron** [comment]
I got mine to work, I will get my code pushed up to my repo in the morning, but I'm going to go pass out now. It will be for the ender 5 plus, but it should get you into the ball park. 

> Dam it Jim, I'm a Systems Administrator, not a programmer.
> > -this jackamo

**Comment by stsweeney41** [comment]
Awesome, thanks!

**Comment by sgsrtcameron** [comment]
@stsweeney41 https://github.com/sgsrtcameron/Marlin/tree/myconfig/E3-SKR-E3-Turbo-Dual-Z-BLTouch This is for the ender 5 plus, but there should only be a few minor changes for the CR10

**Comment by sgsrtcameron** [comment]
@stsweeney41 Was this able to help you at all? Im in the prosses of clossing down this account and didnt want to remove the repo with that commit if you still needed it but had not forked it yet.

**Comment by stsweeney41** [comment]
I finally turned out to be the extruder jamming up (it was a Titan
extruder)... I changed to a BIQU H2 extruder, and all my problems were
fixed. I am sure with some work I could of got the Titan to work properly,
but I had already decided to go with the lighter H2

Hope this helps

**Comment by SpectrumTech** [comment]
I got tired of fighting it and abandoned the CR10S5 for a while and then life got in the way. I just now getting back to it and it has the same issues as before, with the latest marlin stable and bugfix release. 

The odd thing is that I upgraded my ender 3 to the BTT SKR E3 Turbo board soon after abandoning the issues with the CR10S5 and it works just fine. 

The two printers are nearly identical except the size of course, but I added a second Z stepper to the Ender 3, all the same accessories and everything works exactly as it should. I even have tried the ender 3 firmware as is on the CR10S5 as is. The only glaring differences at this point is the CR10S5 has a 12V supply and the Ender 3 is 24V and the Y stepper is larger on the CR10S5. I have a silicone bed heater driven by an SSR on the CR10S5, so the bed current draw shouldn't be a factor. I also completely rewired all the stepper harnesses with heavier gauge wire and new connectors just to make sure something wasn't amiss there.

My best guess at this point is that this board, combined with Marlin and a 12V supply is causing the issues. I do have the SpreadCycle chopper timing set to CHOPPER_TIMING CHOPPER_DEFAULT_12V, since that's the only setting I know of in marlin that would be related. I put a DMM on the supply output and it is rock solid while everything that will work is on and in motion, so I don't think its dirty power. I guess I could put an O'scope on it and check for noise/ripple.  I think at this point I'll go through it and swap everything to 24V, which should be the power supply, a few fans and the extruder heater cartridge. Aside from trying to reverse engineer marlin and how it handles stepper drivers, it's about all there is left to try!
